CMakeFilters.cmake

```diff
@@ -152,14 +152,14 @@ if (HDF5_ENABLE_SZIP_SUPPORT)
       if (NOT SZIP_FOUND)
         find_package (SZIP) # Legacy find
       endif ()
-      if (SZIP_FOUND)
-        set (H5_HAVE_FILTER_SZIP 1)
-        set (H5_HAVE_SZLIB_H 1)
-        set (H5_HAVE_LIBSZ 1)
-        set (SZIP_INCLUDE_DIR_GEN ${SZIP_INCLUDE_DIR})
-        set (SZIP_INCLUDE_DIRS ${SZIP_INCLUDE_DIRS} ${SZIP_INCLUDE_DIR})
-        set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ${SZIP_LIBRARIES})
-      endif ()
+    endif ()
+    if (SZIP_FOUND)
+      set (H5_HAVE_FILTER_SZIP 1)
+      set (H5_HAVE_SZLIB_H 1)
```

Comment:
Actually these first three set routines should be collated into the last if (SZIP_FOUND) block to have one source of these settings.